addons/web/static/src/xml/base.xml

```diff
@@ -1086,33 +1086,48 @@
         </t>
     </div>
 </div>
-<div t-name="FieldBinaryFileUploader.files" class="oe_attachments" aria-atomic="true">
-    <t t-if="widget.mode === 'readonly'">
-        <div t-foreach="widget.value.data" t-as="file">
-            <a target="_blank" t-att-href="widget.metadata[file.id].url" t-attf-title="#{(file.data.name || file.data.filename) + (file.data.date?' \n('+file.data.date+')':'' )}"><t t-raw="file.data.name || file.data.filename"/></a>
-        </div>
-    </t>
-    <t t-else="1">
-        <div t-foreach="widget.value.data" t-as="file" class="oe_attachment">
-            <t t-if="!file.data.upload">
-                <div>
-                    <a href="#" class="fa fa-times float-right oe_delete pl-1" title="Delete this file" aria-label="Delete this file" t-att-data-id="file.data.id"/>
-                    <t t-raw="file.data.name || file.data.filename"/>
+
+<t t-name="AttachmentPreview">
+    <t t-set="type" t-value="mimetype and mimetype.split('/').shift()"/>
+    <t t-set="editable" t-value="widget.mode === 'edit'"/>
+    <div t-attf-class="o_attachment o_attachment_many2many #{ editable ? 'o_attachment_editable' : '' } #{file.upload ? 'o_attachment_uploading' : ''}" t-att-title="file.name">
+        <div class="o_attachment_wrap">
+            <t t-set="ext" t-value="file.name.split('.').pop()"/>
```

Comment:
why do you need the extension ? Are you sure this is the best way to determine it ?

Reply:
for showing the file extension inside the attachment. same like chatter attachment
see here: https://drive.google.com/a/odoo.com/file/d/1nercSmYyXJie9QUEv2ty1EgAgdgbKTrF/view?usp=drivesdk